R/special_orthogonal_group.R

```diff
@@ -89,34 +89,30 @@ SpecialOrthogonalGroup <- setRefClass("SpecialOrthogonalGroup",
       if (.self$n == 3) {
         levi.civita.symbol <- array(data = c(
           c(0, 0, 0),
-          c(0, 0, 1),
-          c(0, -1, 0),
           c(0, 0, -1),
-          c(0, 0, 0),
-          c(1, 0, 0),
           c(0, 1, 0),
+          c(0, 0, 1),
+          c(0, 0, 0),
           c(-1, 0, 0),
+          c(0, -1, 0),
+          c(1, 0, 0),
           c(0, 0, 0)
         ), dim = c(3, 3, 3)
         )
 
-        basis.vec.1 <- ToNdarray(array(c(1, 0, 0) * n.vecs), to.ndim = 2)
-        basis.vec.2 <- ToNdarray(array(c(0, 1, 0) * n.vecs), to.ndim = 2)
-        basis.vec.3 <- ToNdarray(array(c(0, 0, 1) * n.vecs), to.ndim = 2)
-
         ApplyLeviCivitaSymbol <- function(vec) {
           array(cbind(
           vec %*% levi.civita.symbol[1, , ],
           vec %*% levi.civita.symbol[2, , ],
           vec %*% levi.civita.symbol[3, , ]),
-          dim = c(3, 3))
+          dim=c(3,3))
         }
 
-        cross.prod.1 <- vec %*% ApplyLeviCivitaSymbol(basis.vec.1)
-        cross.prod.2 <- vec %*% ApplyLeviCivitaSymbol(basis.vec.2)
-        cross.prod.3 <- vec %*% ApplyLeviCivitaSymbol(basis.vec.3)
+        cross.prod.1 <- vec %*% ApplyLeviCivitaSymbol(array(c(1,0,0)))
+        cross.prod.2 <- vec %*% ApplyLeviCivitaSymbol(array(c(0,1,0)))
+        cross.prod.3 <- vec %*% ApplyLeviCivitaSymbol(array(c(0,0,1)))
 
-        skew.mat <- array(c(cross.prod.1,
+        skew.mat <- -array(c(cross.prod.1,
```

Comment:
maybe you don't need the - sign here, if you put the levi civita symbol back as it was?